### PR TITLE
Preselected currencies for accountplans with selected currency

### LINKED ISSUE #349: Currency preselected for supplier makes trouble in journal.edit + punching of currency
- [x] When currency is preselected for the supplier are debet and credit fields in journal.edit disabled. 
They should not be disabled. 
- [x] In journal.edit, if the supplier has preselected currency, do show this currency immediately if supplier is selected. If the voucher already has some currency, do not edit this, else recalculate currency for default. 
- [x] Do always print both credit and debit currency fields, when currency is selected.

### PR DESCRIPTION
**Pull request by Baxxx**
Closes #349

### CODE REVIEW

File: inc/javascript.inc.php [hunk 1]
```diff
@@ -357,26 +324,31 @@ function calculateFromForeignAmount(element, in_flag) {
...
  */
-function allowOnlyCreditOrDebit(element, credit_or_debit) {
-  var in_amount    = element.parentElement.parentElement.getElementsByTagName('input')[0];
-  var out_amount   = element.parentElement.parentElement.getElementsByTagName('input')[1];
-  var conv_rate    = document.getElementsByName('voucher.ForeignConvRate')[0];
-  var currency     = document.getElementsByName('voucher.ForeignCurrencyID')[0];
-  var f_in_amount  = element.parentElement.parentElement.getElementsByTagName('input')[2];
-  var f_out_amount = element.parentElement.parentElement.getElementsByTagName('input')[3];
+function allowOnlyCreditOrDebit(element, credit_or_debit, from_currency) {
+  if(from_currency) {
```
Comment (Baxxx) [inline inc/javascript.inc.php]: Investigate if we need `<div class='currency_fields'>`. If not, simplify: add the class to the inputs.

File: modules/voucher/model/voucherinput.class.php [hunk 2]
```diff
@@ -106,15 +106,18 @@ function __construct($args) {
 
                 $hash = $_lib['convert']->Amount(array('value'=>$args['voucher_ForeignConvRate']));
                 $this->ForeignConvRate = $hash['value'];
-                $hash = $_lib['convert']->Amount(array('value'=>$args['voucher_ForeignAmount']));
-                $this->ForeignAmount = $hash['value'];
+
+                $ForeignAmountIn  = $_lib['convert']->Amount(array('value'=>$args['voucher_ForeignAmountIn']));
+                $ForeignAmountOut = $_lib['convert']->Amount(array('value'=>$args['voucher_ForeignAmountOut']));
+                $this->ForeignAmount = ($ForeignAmountIn['value'] > 0 ? $ForeignAmountIn['value'] : $ForeignAmountOut['value']);
```
Comment (Baxxx) [inline modules/voucher/model/voucherinput.class.php]: This should be saved as positive.

File: modules/voucher/model/voucherinput.class.php [hunk 3]
```diff
@@ -106,15 +106,18 @@ function __construct($args) {
...
             } else {
                 $this->ForeignCurrencyID = "";
                 $this->ForeignConvRate = 0;
                 $this->ForeignAmount = 0;
             }
         }
 
+        $this->InOrOut = ($args['voucher_AmountIn'] > 0 ? 'in' : ($args['voucher_AmountOut'] > 0 ? 'out' : ($args['voucher_ForeignAmountIn'] > 0 ? 'in' : ($args['voucher_ForeignAmountOut'] > 0 ? 'out' : ''))));
```
Comment (Baxxx) [inline modules/voucher/model/voucherinput.class.php]: Use `x != 0`

File: modules/voucher/model/voucherinput.class.php (hunk 2 above)
Comment (Baxxx) [reply modules/voucher/model/voucherinput.class.php]: Investigate if we need ForeignAmount to be positive or negative at some point.

File: modules/voucher/model/vouchergui.class.php [hunk 4]
```diff
@@ -284,21 +284,29 @@ function update_journal_button_line($voucher, $VoucherPeriod, $JournalID, $Vouch
...
                 if($voucher->DisableAutoVat !=1 )
                 {
-                    $html .= $_lib['form3']->button(array('url' => "$MY_SELF&amp;voucher.VoucherPeriod=$voucher->VoucherPeriod&amp;voucher.VoucherDate=$voucher->VoucherDate&amp;voucher.JournalID=$JournalID&amp;voucher.VoucherID=$voucher->VoucherID&amp;VoucherType=$VoucherType&amp;type=$type&amp;action_voucher_delete=1&amp;view_mvalines=$view_mvalines&amp;view_linedetails=$view_linedetails", 'name'=>'<img src="/lib/icons/trash.gif">', 'confirm' => 'Vil du virkelig slette linjen?'));
+                    if($button == 'delete') {
+                        $html .= $_lib['form3']->button(array('url' => "$MY_SELF&amp;voucher.VoucherPeriod=$voucher->VoucherPeriod&amp;voucher.VoucherDate=$voucher->VoucherDate&amp;voucher.JournalID=$JournalID&amp;voucher.VoucherID=$voucher->VoucherID&amp;VoucherType=$VoucherType&amp;type=$type&amp;action_voucher_delete=1&amp;view_mvalines=$view_mvalines&amp;view_linedetails=$view_linedetails", 'name'=>'<img src="/lib/icons/trash.gif">', 'confirm' => 'Vil du virkelig slette linjen?'));
+                    }
                 }
                 if($voucher->DisableAutoVat !=1 )
```
Comment (Baxxx) [inline modules/voucher/model/vouchergui.class.php]: Remove

Comment (Baxxx) [inline modules/voucher/model/vouchergui.class.php]: Merge with upper conditional

**Review by sls-lnluksa — APPROVED** [review]
Final review with: @stoivo

File: modules/exchange/model/exchange.class.php [hunk 5]
```diff
@@ -362,5 +302,19 @@ static function googleExchangeRateUrl($amount, $currency, $exchangeIn)
         $googleQuery = urlEncode( $googleQuery );
         return 'http://www.google.com/search?q=' . $googleQuery;
     }
+
+    function getCurrenciesInJS() {
+      $currency_js = '';
+      $currencies = self::getActiveCurrencies();
+      if (!empty($currencies)) {            
+        $currency_js = '<script type="text/javascript">';
```
Comment (Baxxx) [inline modules/exchange/model/exchange.class.php]: Use `.=`

**Review by stoivo — APPROVED** [review]

**Pull request merged**